ci-operator/step-registry/aws/deprovision/kms-key/aws-deprovision-kms-key-commands.sh

```diff
@@ -7,14 +7,23 @@ set -o pipefail
 export AWS_SHARED_CREDENTIALS_FILE="${CLUSTER_PROFILE_DIR}/.awscred"
 REGION="${LEASED_RESOURCE}"
 
-key_id=$(head -n 1 ${SHARED_DIR}/aws_kms_key_id)
-pending_days=7
+function delete_kms_key()
+{
+    local key_id=$1
+    local pending_days=7
 
-if [[ "${key_id}" == "" ]]; then
-    echo "ERROR: KMS key is empty."
-    exit 1
-fi
+    echo "KMS key $key_id will be delete after $pending_days days."
+    aws --region $REGION kms schedule-key-deletion --key-id "${key_id}" --pending-window-in-days $pending_days
+}
 
-
-echo "KMS key $key_id will be delete after $pending_days days."
-aws --region $REGION kms schedule-key-deletion --key-id "${key_id}" --pending-window-in-days $pending_days
+ret=0
+for key_id_file in ${SHARED_DIR}/aws_kms_key_id*;
+do
+    echo "Checking ${key_id_file}"
+    key_id=$(head -n 1 ${key_id_file})
+    if [[ "${key_id}" == "" ]]; then
+        echo "ERROR: key_id can not be found in ${key_id_file}"
+        ret=$((ret+1))
```

Comment:
Seem like we did not use `ret`

Reply:
If the key id are not found, we should be continue the loop to the next item, but not continue to run `delete_kms_key`, right?

Reply:
Ah, right, let me update